src/realm/obj.cpp

```diff
@@ -237,6 +237,35 @@ CollectionListPtr CollectionList::get_collection_list(size_t ndx) const
                                             coll_type);
 }
 
+void CollectionList::remove(size_t ndx)
+{
+    ensure_created();
+    REALM_ASSERT(m_key_type == type_Int);
+    auto int_keys = static_cast<BPlusTree<Int>*>(m_keys.get());
+    if (ndx < int_keys->size()) {
+        auto btree_index = int_keys->find_first(int_keys->get(ndx));
+        if (btree_index != realm::not_found) {
+            int_keys->erase(btree_index);
+            m_refs.erase(btree_index);
```

Comment:
The sub-tree that the deleted ref points to in the file should be freed. (`Array::destroy_deep()`)

Reply:
done